### PR TITLE
[MOBILE-385] update react-native sample to latest

### PR DESCRIPTION
**Pull request by crow**
### What do these changes do?
Updates sample to use react-native 0.57.5

### Why are these changes necessary?
Keep things fresh

### How did you verify these changes?
Ran both samples (both targets in iOS as well)

### CODE REVIEW

File: sample/AirshipSample/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,15 +1,14 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.urbanairship.sample">
+    package="com.airshipsample">
```
Comment (BrianBatchelder) [inline sample/AirshipSample/android/app/src/main/AndroidManifest.xml]: Why did you make this change? Do we own `airshipsample.com`?

File: sample/AirshipSample/android/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,5 @@
-#Thu Nov 08 16:40:30 PST 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
```
Comment (BrianBatchelder) [inline sample/AirshipSample/android/gradle/wrapper/gradle-wrapper.properties]: Why did you downgrade gradle?

File: sample/AirshipSample/ios/AirshipSample Carthage-Info.plist [hunk 3]
```diff
@@ -36,9 +38,7 @@
 		</dict>
 	</dict>
 	<key>NSLocationWhenInUseUsageDescription</key>
-	<string>Urban Airship React Native sample location, when in use</string>
-	<key>NSLocationAlwaysUsageDescription</key>
-	<string>Urban Airship React Native sample location</string>
+	<string></string>
```
Comment (BrianBatchelder) [inline sample/AirshipSample/ios/AirshipSample Carthage-Info.plist]: Seems odd to have the key with an empty string value. Why not just remove the key and value?

File: sample/AirshipSample/ios/AirshipSample Cocoapods-Info.plist [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+		<dict>
+			<key>localhost</key>
+			<dict>
+				<key>NSExceptionAllowsInsecureHTTPLoads</key>
+				<true/>
+			</dict>
+		</dict>
+	</dict>
+	<key>NSLocationWhenInUseUsageDescription</key>
+	<string></string>
```
Comment (BrianBatchelder) [inline sample/AirshipSample/ios/AirshipSample Cocoapods-Info.plist]: same as above.

File: sample/AirshipSample/ios/AirshipSample-tvOS/Info.plist [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	<key>UISupportedInterfaceOrientations</key>
+	<array>
+		<string>UIInterfaceOrientationPortrait</string>
+		<string>UIInterfaceOrientationLandscapeLeft</string>
+		<string>UIInterfaceOrientationLandscapeRight</string>
+	</array>
+	<key>UIViewControllerBasedStatusBarAppearance</key>
+	<false/>
+	<key>NSLocationWhenInUseUsageDescription</key>
+	<string></string>
```
Comment (BrianBatchelder) [inline sample/AirshipSample/ios/AirshipSample-tvOS/Info.plist]: Same as above.

File: sample/AirshipSample/ios/AirshipSample/AppDelegate.h [hunk 6]
```diff
@@ -1,10 +1,8 @@
 /**
  * Copyright (c) 2015-present, Facebook, Inc.
```
Comment (BrianBatchelder) [inline sample/AirshipSample/ios/AirshipSample/AppDelegate.h]: Why is our sample code copyright "Facebook"?

Comment (crow) [reply sample/AirshipSample/ios/AirshipSample/AppDelegate.h]: autogenerated

Comment (BrianBatchelder) [reply sample/AirshipSample/ios/AirshipSample/AppDelegate.h]: Well, thanks Facebook!

File: sample/AirshipSample/ios/AirshipSample Cocoapods-Info.plist (hunk 4 above)
Comment (crow) [reply sample/AirshipSample/ios/AirshipSample Cocoapods-Info.plist]: fixed the descriptions

File: sample/AirshipSample/ios/AirshipSample Carthage-Info.plist (hunk 3 above)
Comment (crow) [reply sample/AirshipSample/ios/AirshipSample Carthage-Info.plist]: broke the single info plist in two to account for the two targets, re-added these keys

File: android/build.gradle [hunk 7]
```diff
@@ -13,5 +13,5 @@ android {
 dependencies {
     implementation 'com.facebook.react:react-native:[0.40,)'
     implementation 'com.urbanairship.android:urbanairship-fcm:9.5.6'
-    implementation 'com.google.firebase:firebase-core:16.0.5'
+    implementation 'com.google.firebase:firebase-messaging:17.3.4'
```
Comment (rlepinski) [inline android/build.gradle]: remove this

File: sample/scripts/run_ci_tasks.sh [hunk 8]
```diff
@@ -47,13 +47,13 @@ if [[ "$BITRISE_SOURCE_DIR" != "" ]]; then
...
     REPO_PATH=`dirname "${0}"`/../../
 fi
+
 cd $REPO_PATH
 pwd
 npm link
 cd sample/AirshipSample
-npm link urbanairship-react-native
 npm install
-react-native link urbanairship-react-native
```
Comment (rlepinski) [inline sample/scripts/run_ci_tasks.sh]: we already have it linked in the sample. Since we changed `compile` to `implementation`, it adds the module twice

**Comment by rlepinski** [comment]
👍

**Pull request merged**